sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -868,6 +868,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
     case null => Nil
     case None => Nil
     case Some(null) => Nil
+    case Some(table: CatalogTable) =>
```

Comment:
This is needed to represent the CatalogTable properly if it is presented. (Before the fix it calls CatalogTable.toString which produces a bunch of information with multiple lines.)

I feel it could be something be generalized, but I don't know we have other existing cases to handle.

Reply:
It's already the case for batch queries and I'd suggest to do it in a separated PR if we want to

Reply:
Ah OK. Didn't realize logical nodes for batch also have such case. I agree it's better to deal with separate PR.